### PR TITLE
feat: [MAM-322] onboarding follow categories API

### PR DESCRIPTION
**Pull request by felipecsl**
## Summary

This is a new API to be called during the onboarding flow for new Mammoth users to suggest influential and prolific users to follow based on a few predetermined categories. User suggestions are pre-populated in a YAML file at `app/lib/onboarding/onboarding_categories_production.yml` (there's a separate one for tests).
In order to change the recommendations, this file needs to be modified and the backend deployed.

## Testing

wrote unit tests

## Rollout plan

- [x] For each account in the yml file, we need to run `ResolveAccountWorker.perform_async(handle)` in order to populate those accounts in the DB, otherwise we won't have account data to return in the response

### CODE REVIEW

File: app/controllers/api/v1/onboarding_follow_recommendations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  private
+
+  def onboarding_follow_recommendations
+    categories = YAML.load_file(yaml_file_location)
+    categories.map do |category|
+      Onboarding::FollowRecommendationCategory.new(
+        name: category['name'],
+        theme_color: category['color'],
+        # If any of YML accounts are invalid or not found in the database, we'll omit them from the response
+        accounts: category['accounts'].filter_map { |a| Account.find_remote(*username_and_domain(a)) }
```
Comment (sdubinsky) [inline app/controllers/api/v1/onboarding_follow_recommendations_controller.rb:22]: smart

File: app/lib/onboarding/follow_recommendation_category.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Represents a grouping of suggested accounts that can be followed by new users on moth.social.
+# Categories are grouped by theme, eg.: tech, news, art, etc.
```
Comment (sdubinsky) [inline app/lib/onboarding/follow_recommendation_category.rb:4]: does this theme show up in the app anywhere, or is it just for our own knowledge?

File: app/controllers/api/v1/onboarding_follow_recommendations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Not to be confused with [Api::V1::Accounts::FollowRecommendationsController].
+# This API suggests a hardcoded list of follows for new users while they are going through the
+# onboarding flow after signing up for a new account.
+# FollowRecommendationsController is used, instead, to suggest new follows to existing users based on
```
Comment (sdubinsky) [inline app/controllers/api/v1/onboarding_follow_recommendations_controller.rb:6]: I think you're missing a bit here - shouldn't it read "If the user is already following people, FollowRecommendationsController is used, instead..."?

**Review by sdubinsky — APPROVED** [review]

**Comment by sdubinsky** [comment]
looks good!  Left a couple questions for my own knowledge.

File: app/lib/onboarding/follow_recommendation_category.rb (hunk 2 above)
Comment (felipecsl) [reply app/lib/onboarding/follow_recommendation_category.rb:4]: it doesnt, actually I think @rileyhCode wanted to get it removed sinde its not gonna be used by the client at all. I'll confirm with him

File: app/controllers/api/v1/onboarding_follow_recommendations_controller.rb (hunk 3 above)
Comment (felipecsl) [reply app/controllers/api/v1/onboarding_follow_recommendations_controller.rb:6]: I think these 2 APIs are used in different contexts in different parts of the app. You could call `FollowRecommendationsController` even if you're not following anyone, it's just that the suggestions won't be that great hahah

Comment (sdubinsky) [reply app/controllers/api/v1/onboarding_follow_recommendations_controller.rb:6]: ohh, got it

**Pull request merged**